### PR TITLE
Adding daily upgrade test script.

### PR DESCRIPTION
**Pull request by hklai**

### CODE REVIEW

File: prow/daily-upgrade-tests.sh [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+pushd istio
+#from now on we are in ${GOPATH}/src/istio.io/istio dir
+
+git checkout $ISTIO_SHA
+source "prow/mason_lib.sh"
+source "prow/cluster_lib.sh"
+trap cleanup EXIT
+
+# Download envoy and go deps
+make init
```
Comment (hklai) [inline prow/daily-upgrade-tests.sh]: I suppose this is not needed for this test,.

Comment (rkpagadala) [reply prow/daily-upgrade-tests.sh]: yes

**Comment by hklai** [comment]
/hold cancel

daily-upgrade-tests.sh did run correctly. The job currently fails due to 503s during upgrade. It will be addressed separately.

@rkpagadala PTAL again.

**Pull request merged**